### PR TITLE
add custom Delims support

### PR DESCRIPTION
**Pull request by shenshouer**
add custom delims support, for example:

```
...
r := gin.Default()
r.Delims("{[{", "}]}")
r.LoadHTMLGlob("/path/to/templates"))
...
...
```

### CODE REVIEW

**Comment by appleboy** [comment]
Wrong branch and missing testing.

**Comment by shenshouer** [comment]
@appleboy How can I do for this PR?

**Comment by appleboy** [comment]
@Saser I changed the base branch. You just add testing code on your branch and push commits.

**Comment by shenshouer** [comment]
@appleboy Need I add testing code on the master branch , then push commits ? Or others?

**Comment by appleboy** [comment]
@shenshouer Push your commits in your `master` branch.

File: gin.go [hunk 1]
```diff
@@ -121,21 +123,27 @@ func (engine *Engine) allocateContext() *Context {
...
+func (engine *Engine) Delims(left, right string) *Engine {
+	engine.delims = render.Delims{left, right}
+	return engine
+}
+
 func (engine *Engine) LoadHTMLGlob(pattern string) {
 	if IsDebugging() {
-		debugPrintLoadTemplate(template.Must(template.ParseGlob(pattern)))
-		engine.HTMLRender = render.HTMLDebug{Glob: pattern}
+		// debugPrintLoadTemplate(template.Must(template.ParseGlob(pattern)))
```
Comment (tboerger) [inline gin.go]: No need for this comments code

File: gin_test.go [hunk 2]
```diff
@@ -8,11 +8,60 @@ import (
 	"reflect"
 	"testing"
 
+	"net/http"
+
```
Comment (tboerger) [inline gin_test.go]: Why all these blank lines?

Comment (shenshouer) [inline gin_test.go]: Sorry for the miss inspection, this was auto import package and format style by the vscode.

Comment (appleboy) [inline gin_test.go]: please remove the empty line for import native package.

File: gin.go (hunk 1 above)
Comment (appleboy) [inline gin.go]: @shenshouer Please remove this comments.

**Comment by appleboy** [comment]
@tboerger Please review again.

**Comment by tboerger** [comment]
Looks like the commits ate messed up, you need to rebase properly.

**Comment by appleboy** [comment]
@tboerger NO. It doesn't matter about previous commits. We can squash and merge this PR.

**Comment by abrander** [comment]
How can we get this merged?

**Comment by appleboy** [comment]
@shenshouer Thanks for your contribution.

**Pull request merged**